scvi/model/_condscvi.py

```diff
@@ -128,34 +128,72 @@ def get_vamp_prior(
         )
         key = labels_state_registry.original_key
         mapping = labels_state_registry.categorical_mapping
-        for ct in range(self.summary_stats.n_labels):
-            # pick p cells
-            local_indices = np.random.choice(
-                np.where(adata.obs[key] == mapping[ct])[0], p
+
+        scdl = self._make_data_loader(adata=adata, batch_size=p)
+
+        mean = []
+        var = []
+        for tensors in scdl:
+            x = tensors[REGISTRY_KEYS.X_KEY]
+            y = tensors[REGISTRY_KEYS.LABELS_KEY]
+            out = self.module.inference(x, y)
+            mean_, var_ = out["qz_m"], out["qz_v"]
+            mean += [mean_.cpu()]
+            var += [var_.cpu()]
+
+        mean_cat, var_cat = np.array(torch.cat(mean)), np.array(torch.cat(var))
+        adata.obsm["X_CondSCVI"] = mean_cat
+
+        for ct in range(self.summary_stats["n_labels"]):
+            local_indices = np.where(adata.obs[key] == mapping[ct])[0]
+            sub_adata = adata[local_indices, :].copy()
+            if "overclustering_vamp" not in sub_adata.obs.columns:
+                from scanpy.pp import neighbors
+                from scanpy.tl import leiden
```

Comment:
this should look more like this:

https://github.com/scverse/scvi-tools/blob/4c4bd29c279c95fc85ab3ce4603a310debb57be0/scvi/model/utils/_mde.py#L53-L56